pkg/config/labels/instance.go

```diff
@@ -140,25 +140,25 @@ func validateTagKey(k string) error {
 }
 
 func (i Instance) String() string {
-	labels := make([]string, 0, len(i))
-	for k, v := range i {
-		if len(v) > 0 {
-			labels = append(labels, fmt.Sprintf("%s=%s", k, v))
-		} else {
-			labels = append(labels, k)
-		}
-	}
-	sort.Strings(labels)
+	// Ensure stable ordering
+	keys := slices.Sort(maps.Keys(i))
 
-	var buffer bytes.Buffer
+	var buffer strings.Builder
+	// Assume each kv pair is roughly 10 characters. We could be under or over, this is just a guess to optimize
```

Comment:
I am not sure this assumption is right, in real many software has their long domain suffix added to the key

like `app.kubernetes.io/name`:
`cluster.x-k8s.io/provider:`

Reply:
I agree, figured it was better to underestimate though?

Reply:
updated it